src/helpers/validateFormData.ts

```
@@ -0,0 +1,35 @@
+  const errors: Record<string, string> = {};
+
+  try {
+    // Attempt to parse the form data using the provided schema.
+    formSchema.parse(inputs);
+  } catch (err: any) {
+    // If a validation error is caught, extract field-specific errors.
+    const { fieldErrors } = err.flatten();
+
+    // Iterate through the field errors and store the first error for each field.
+    for (const key in fieldErrors) {
+      if (fieldErrors.hasOwnProperty(key)) {
```

Biome (1.9.4)

[error] Do not access Object.prototype method 'hasOwnProperty' from target object.

It's recommended using Object.hasOwn() instead of using Object.hasOwnProperty().
See MDN web docs for more details.

(lint/suspicious/noPrototypeBuiltins)